### PR TITLE
NC to texture atlas

### PR DESCRIPTION
**Pull request by kmilo9999**
- Added logic to convert nc erddap data to png texture atlas for web and desktop volume viewer. 
- Rename variables

### CODE REVIEW

**Review by gtdang — COMMENTED** [review]
I think this looks ok. Had a few questions before approval.

File: volume_data_converter/create_volume_viewer_data.py [hunk 1]
```diff
@@ -91,30 +94,25 @@ def create_osom_data(
...
         if "s_rho" not in data_dimensions:
             raise Exception("current dataset does not support elevation layers")
     else:
-        # no elevation data. Map the data to a empty block of 'verticalayers' layers.
-        # Our current cases are surface and bottom. They map to layer 0 and 14 respectively
-        new_data = np.ma.zeros(
-            (data.shape[0], verticalLevels, data.shape[1], data.shape[2]),
-            dtype=data.dtype,
+        # Single layer case (bottom, surface). Create 3D matrix for each time frame with the
+        # current data and mask the no relevant ( no data ) slices.
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply volume_data_converter/create_volume_viewer_data.py]: done

File: volume_data_converter/create_volume_viewer_data.py [hunk 2]
```diff
@@ -42,6 +44,7 @@ def create_osom_data(
     data_descriptor = parameters["data_descriptor"]
     time_frames = parameters.get("time_frames", None)
     layer = parameters.get("layer", "all")
+    to_texture_atlas = parameters.get("to_texture_atlas", False)
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply volume_data_converter/create_volume_viewer_data.py:51]: I updated the docustring

File: volume_data_converter/create_volume_viewer_data.py [hunk 3]
```diff
@@ -196,53 +194,87 @@ def create_osom_data(
...
+                                t_new = np.interp(query_depths, domain, mapped_values)
+                                if layer == "surface":
+                                    out_data[-20:, x, y] = np.round(
+                                        t_new[0:20], decimals=4
+                                    )
+                                else:
+                                    # bottom
+                                    out_data[:20, x, y] = np.round(
+                                        t_new[0:20], decimals=4
+                                    )
```
[the remark this answers is not in the document]
Comment (kmilo9999) [reply volume_data_converter/create_volume_viewer_data.py:217]: Those are kind of magic numbers I used. In a single layer case, there is no interpolation, just the same top or bottom layer repeated a few times to make it look good.

[the remark this answers is not in the document]
Comment (gtdang) [reply volume_data_converter/create_volume_viewer_data.py:217]: Ok sounds good!

**Review by gtdang — APPROVED** [review]
Looks good!

**Pull request merged**